rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LRAParticipantRecord.java

```diff
@@ -114,11 +116,13 @@ public LRAParticipantRecord() {
                 if (parseException[0] != null) {
                     String errorMsg = LRALogger.i18nLogger.error_invalidCompensator(lra.getId(), parseException[0].getMessage(), linkURI);
                     LRALogger.logger.error(errorMsg);
+                    trace_progress(errorMsg);
```

Comment:
First, I'm fine this being in the code as you prefer it. This is just I would like just express my opinion.
This is kind of like debugging with `System.out.println` and that way it slows down the processing in general. Hopefully the JIT will cover it.

Reply:
Your statement is incorrect: trace_progress is coded as `if (LRALogger.logger.isTraceEnabled()) {...}` and I make sure I only call `trace_progress` with static strings from the heap, ie the only performance hit is an extra frame.

Reply:
I think my statement is quite correct. The performance hit is an extra frame to the stack and an if operation to verify the trace is enabled. In the hot code path it's not a great strategy. If you stress this as an benefit for other reasons, no problem.

Reply:
The logical conclusion of your argument is to remove all of our debug and trace level logging and we have plenty of that.

A single frame is minimal and if you really were worried about performance then there are loads of other opportunities to improve the throughput, besides the calls are only ever made with strings from the heap. If we find sufficient uptake of LRA then we can begin looking at sprucing up the code base, until then maintainability is more important.

Is this an impasse for which we need an adjudicator?

Reply:
Do also notice that the module is I/O bound and most of the trace logging happens in response to a state change which needs to be persisted, ie the performance hit of an extra frame is dwarfed by other costs.

Reply:
Sorry for not being clear. This is not an impasse. I'm fine with the PR to be merged and I've approved it already.

Just I have a different perspective on coding practices. For example, I would be supporting and idea to find the places where to emit open tracing events. While I would not track the processing with log trace messages.
The most of the log are written at places of a non standard processing. That's how it should be by me. Not logging on hot code path but in strange and exceptional situations.

Never-mind, I don't want to both you with my arguments. I feel I need to express my perspective which can lead to some discussion on the topic. I don't oppose merging this PR.

Reply:
Let's wait until we have an opentracing or a bytecode/byteman based solution before throwing out trace logging.
Until that work get's done we need to be able trace the progress of LRAs and compensators.